### PR TITLE
[feature](information_schema) add `rowsets` table into information_s…

### PR DESCRIPTION
**Pull request by BePPPower**
…chema

# Proposed changes

Issue Number: close #xxx

## Problem Summary:
In some cases, we would like to check the status of `rowsets`. So I add the `rowsets` table into information_schema database .
The table schema of `rowsets` is:
```
MySQL [(none)]> desc information_schema.rowsets;
+------------------------+------------+------+-------+---------+-------+
| Field                  | Type       | Null | Key   | Default | Extra |
+------------------------+------------+------+-------+---------+-------+
| BACKEND_ID             | BIGINT     | Yes  | false | NULL    |       |
| ROWSET_ID              | VARCHAR(*) | Yes  | false | NULL    |       |
| TABLET_ID              | BIGINT     | Yes  | false | NULL    |       |
| ROWSET_NUM_ROWS        | BIGINT     | Yes  | false | NULL    |       |
| TXN_ID                 | BIGINT     | Yes  | false | NULL    |       |
| NUM_SEGMENTS           | BIGINT     | Yes  | false | NULL    |       |
| START_VERSION          | BIGINT     | Yes  | false | NULL    |       |
| END_VERSION            | BIGINT     | Yes  | false | NULL    |       |
| INDEX_DISK_SIZE        | BIGINT     | Yes  | false | NULL    |       |
| DATA_DISK_SIZE         | BIGINT     | Yes  | false | NULL    |       |
| CREATION_TIME          | BIGINT     | Yes  | false | NULL    |       |
| OLDEST_WRITE_TIMESTAMP | BIGINT     | Yes  | false | NULL    |       |
| NEWEST_WRITE_TIMESTAMP | BIGINT     | Yes  | false | NULL    |       |
+------------------------+------------+------+-------+---------+-------+
```

and then we can search `rowsets` info from `rowsets` table, like this:
```
MySQL [(none)]> select * from information_schema.rowsets where BACKEND_ID = 10004 limit 10;
+------------+--------------------------------------------------+-----------+-----------------+--------+--------------+---------------+-------------+-----------------+----------------+---------------+------------------------+------------------------+
| BACKEND_ID | ROWSET_ID                                        | TABLET_ID | ROWSET_NUM_ROWS | TXN_ID | NUM_SEGMENTS | START_VERSION | END_VERSION | INDEX_DISK_SIZE | DATA_DISK_SIZE | CREATION_TIME | OLDEST_WRITE_TIMESTAMP | NEWEST_WRITE_TIMESTAMP |
+------------+--------------------------------------------------+-----------+-----------------+--------+--------------+---------------+-------------+-----------------+----------------+---------------+------------------------+------------------------+
|      10004 | 02000000000000994847fbd41a42297d7c7a57d3bcb46f8c |     10771 |           66850 |      6 |            1 |             3 |           3 |            2894 |         688855 |    1659964582 |             1659964581 |             1659964581 |
|      10004 | 020000000000008d4847fbd41a42297d7c7a57d3bcb46f8c |     10771 |           66850 |      2 |            1 |             2 |           2 |            2894 |         688855 |    1659964575 |             1659964574 |             1659964574 |
|      10004 | 02000000000000894847fbd41a42297d7c7a57d3bcb46f8c |     10771 |               0 |      0 |            0 |             0 |           1 |               0 |              0 |    1659964567 |             1659964567 |             1659964567 |
|      10004 | 020000000000009a4847fbd41a42297d7c7a57d3bcb46f8c |     10773 |           66639 |      6 |            1 |             3 |           3 |            2897 |         686828 |    1659964582 |             1659964581 |             1659964581 |
|      10004 | 020000000000008e4847fbd41a42297d7c7a57d3bcb46f8c |     10773 |           66639 |      2 |            1 |             2 |           2 |            2897 |         686828 |    1659964575 |             1659964574 |             1659964574 |
|      10004 | 02000000000000884847fbd41a42297d7c7a57d3bcb46f8c |     10773 |               0 |      0 |            0 |             0 |           1 |               0 |              0 |    1659964567 |             1659964567 |             1659964567 |
|      10004 | 02000000000000984847fbd41a42297d7c7a57d3bcb46f8c |     10757 |           66413 |      6 |            1 |             3 |           3 |            2893 |         685381 |    1659964582 |             1659964581 |             1659964581 |
|      10004 | 020000000000008c4847fbd41a42297d7c7a57d3bcb46f8c |     10757 |           66413 |      2 |            1 |             2 |           2 |            2893 |         685381 |    1659964575 |             1659964574 |             1659964574 |
|      10004 | 02000000000000874847fbd41a42297d7c7a57d3bcb46f8c |     10757 |               0 |      0 |            0 |             0 |           1 |               0 |              0 |    1659964567 |             1659964567 |             1659964567 |
|      10004 | 020000000000009c4847fbd41a42297d7c7a57d3bcb46f8c |     10739 |            1698 |      8 |            1 |             3 |           3 |             454 |          86126 |    1659964582 |             1659964582 |             1659964582 |
+------------+--------------------------------------------------+-----------+-----------------+--------+--------------+---------------+-------------+-----------------+----------------+---------------+------------------------+------------------------+
```

## Checklist(Required)

1. Type of your changes:
    - [ ] Improvement
    - [ ] Fix
    - [ ] Feature-WIP
    - [x] Feature
    - [ ] Doc
    - [ ] Refator
    - [ ] Others: 
2. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yiguolei — CHANGES_REQUESTED** [review]
I will review this pr.

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 1]
```diff
@@ -1699,7 +1699,11 @@ private PlanNode createScanNode(Analyzer analyzer, TableRef tblRef, SelectStmt s
                 scanNode = new MysqlScanNode(ctx.getNextNodeId(), tblRef.getDesc(), (MysqlTable) tblRef.getTable());
                 break;
             case SCHEMA:
-                scanNode = new SchemaScanNode(ctx.getNextNodeId(), tblRef.getDesc());
+                if (BackendSchemaScanNode.isBackendSchemaTable(tblRef.getDesc().getTable().getName())) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:1702]: BackendPartitionedSchemaScanNode

File: be/src/exec/schema_scanner/schema_segments_scanner.cpp [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    }
+    *eos = false;
+    return fill_one_row(tuple, pool);
+}
+
+Status SchemaSegmentsScanner::get_all_rowsets() {
+    std::vector<TabletSharedPtr> tablets =
+            StorageEngine::instance()->tablet_manager()->get_all_tablet();
+    for (const auto& tablet : tablets) {
+        TabletMetaSharedPtr tabletMetas = tablet->tablet_meta();
```
Comment (yiguolei) [inline be/src/exec/schema_scanner/schema_segments_scanner.cpp:91]: Is this field useful?

File: be/src/exec/schema_scanner/schema_segments_scanner.cpp [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    if (!_is_init) {
+        return Status::InternalError("used before initialized.");
+    }
+    backend_id_ = state->backend_id();
+    RETURN_IF_ERROR(get_all_rowsets());
+    return Status::OK();
+}
+
+Status SchemaSegmentsScanner::get_next_row(Tuple* tuple, MemPool* pool, bool* eos) {
+    if (!_is_init) {
```
Comment (yiguolei) [inline be/src/exec/schema_scanner/schema_segments_scanner.cpp:69]: 这个实现向量化的版本，不要实现非向量化的版本。

File: be/src/exec/schema_scanner/schema_segments_scanner.cpp [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            rowsets_.emplace_back(rowset);
+        }
+    }
+    return Status::OK();
+}
+
+Status SchemaSegmentsScanner::get_new_segments() {
+    BetaRowsetSharedPtr beta_rowset = std::dynamic_pointer_cast<BetaRowset>(rowsets_[rowsets_idx_]);
+    segments_.clear();
+    RETURN_IF_ERROR(beta_rowset->load_segments(&segments_));
```
Comment (yiguolei) [inline be/src/exec/schema_scanner/schema_segments_scanner.cpp:110]: 这里调用load segments 太重了，会打开一堆文件。
是不是可以只通过rowset meta类似的pb 结构得到。

Comment (yiguolei) [reply be/src/exec/schema_scanner/schema_segments_scanner.cpp:110]: 你看121行，我们只是为了拿到一下PB 结构，感觉没有必要这么大动干戈。

File: fe/fe-core/src/main/java/org/apache/doris/catalog/SchemaTable.java [hunk 5]
```diff
@@ -384,7 +384,22 @@ public class SchemaTable extends Table {
...
                                     .column("COLLATION_CONNECTION", ScalarType.createVarchar(32))
-                                    .column("DATABASE_COLLATION", ScalarType.createVarchar(32)).build())).build();
+                                    .column("DATABASE_COLLATION", ScalarType.createVarchar(32)).build()))
+            .put("segments", new SchemaTable(SystemIdGenerator.getNextId(), "segments", TableType.SCHEMA,
+                            builder().column("BACKEND_ID", ScalarType.createType(PrimitiveType.BIGINT))
+                                    .column("ROWSET_ID", ScalarType.createVarchar(64))
+                                    .column("TABLET_ID", ScalarType.createType(PrimitiveType.BIGINT))
+                                    .column("ROWSET_NUM_ROWS", ScalarType.createType(PrimitiveType.BIGINT))
+                                    .column("TXN_ID", ScalarType.createType(PrimitiveType.BIGINT))
+                                    .column("PARTITION_ID", ScalarType.createType(PrimitiveType.BIGINT))
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/SchemaTable.java:394]: 不要partitionid了

Comment (BePPPower) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/SchemaTable.java]: done

File: be/src/exec/schema_scanner/schema_segments_scanner.cpp (hunk 2 above)
Comment (BePPPower) [reply be/src/exec/schema_scanner/schema_segments_scanner.cpp]: it's not used. done

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java (hunk 1 above)
Comment (BePPPower) [reply fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java]: done

File: be/src/exec/schema_scanner/schema_segments_scanner.cpp (hunk 4 above)
Comment (BePPPower) [reply be/src/exec/schema_scanner/schema_segments_scanner.cpp]: done, we can't lightweight load segments, so we have to give up the segments information.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**